### ISSUE TITLE
Option to provide registry credentials via environment variables

### ISSUE THREAD

**Issue by fkorotkov** [cirruslabs/tart#214]
In an automated environment like CI especially if it's not ephemeral Keychain is not a good option for storing credentials. One will always need to clean the credentials. Credential helpers #167 also has an intention to store things locally.

IMO for one time pushes to registry Tart should have an option to provide credentials via `TART_REGISTRY_USERNAME` and `TART_REGISTRY_PASSWORD`.

**Comment by mortidlo** [comment]
This feature would be really helpful for our project. We use tart in GitLab CI jobs and we have to work with keychains. Using env variables for password/username would be great.

**Comment by edigaryev** [comment]
> IMO for one time pushes to registry Tart should have an option to provide credentials via `TART_REGISTRY_USERNAME` and `TART_REGISTRY_PASSWORD`.

I think this will only support a single registry **or** give out the registry-specific credentials to all registries that Tart connects to, which is not ideal.

> This feature would be really helpful for our project. We use tart in GitLab CI jobs and we have to work with keychains. Using env variables for password/username would be great.

@mortidlo can you elaborate a bit more on the problem you're facing with Tart's usage of Keychain? Is it the Keychain cleaning process, or something else?

**Comment by edigaryev** [comment]
> So for us the most convenient way would be setup environment variables for authentication same way as for example docker does for this one time pushes. So we would just type `tart login ….` inside of job and then push image.

Thanks for the explanation, do you have some link to the documentation about Docker one time pushes?

Also, does:

```
echo -n $REGISTRY_PASSWORD | tart login --username $REGISTRY_USERNAME --password-stdin
```

...work for you?

**Comment by mortidlo** [comment]
Yes login works exactly this way. I think you only need to learn tart store credentials outside of keychain in something similar to .docker.json and use Bearer authentication against registry endpoint.

Edit: I saw you have already merged in #320 and also documented it. Thank you very much.

**Comment by fkorotkov** [comment]
Not sure why this issue was not automatically closed though.
